### PR TITLE
Add SCC related config for OpenShift

### PR DESCRIPTION
**Pull request by concaf**
# Changes

# Submitter Checklist

These are the criteria that every PR should meet, please check them off as you
review them:

- [x] Run `make test lint` before submitting a PR
- [x] Includes [tests](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if functionality changed/added)
- [ ] Includes [docs](https://github.com/tektoncd/community/blob/master/standards.md#principles) (if user facing)
- [x] Commit messages follow [commit message best practices](https://github.com/tektoncd/community/blob/master/standards.md#commit-messages)

_See [the contribution guide](https://github.com/tektoncd/operator/blob/master/CONTRIBUTING.md) for more details._

# Release Notes

### CODE REVIEW

**Comment by concaf** [comment]
implementation being done based on solution 2 here - https://docs.google.com/document/d/1UzEou4FZUs9oMtNhCgnDvBJ28u24DNm6bCCkB5c49IA/edit#

File: pkg/reconciler/openshift/tektonconfig/rbac.go [hunk 1]
```diff
@@ -223,7 +292,89 @@ func (r *rbac) createResources(ctx context.Context) error {
...
+					ObjectMeta: metav1.ObjectMeta{
+						GenerateName:    "pipelines-scc-failure-",
+						Namespace:       n.GetName(),
+						OwnerReferences: []metav1.OwnerReference{r.ownerRef},
+					},
+					EventTime:           metav1.NewMicroTime(time.Now()),
+					Reason:              "RequestedSCCNotFound",
+					Type:                "Warning",
+					Action:              "DefaultSCCUSed",
+					Message:             fmt.Sprintf("SCC '%s' requested in annotation '%s' but could not find SCC, OpenShift Pipelines workloads will not run in this namespace", namespaceSCC, v1alpha1.NamespaceSCCAnnotation),
```
Comment (vdemeester) [inline pkg/reconciler/openshift/tektonconfig/rbac.go:321]: Workloads can run, but with the openshift "default" scc and stuff right ? or the cluster-wide max ?

Comment (concaf) [reply pkg/reconciler/openshift/tektonconfig/rbac.go:321]: yep, changed the message to
```
fmt.Sprintf("SCC '%s' requested in annotation '%s' not found, SCC not updated in the namespace", scc, namespaceSCCAnnotation)
```

**Review by vdemeester — APPROVED** [review]

**Review by jkandasa — APPROVED** [review]

File: pkg/apis/operator/v1alpha1/openshift_platform.go [hunk 2]
```diff
@@ -30,3 +33,15 @@ type PipelinesAsCode struct {
...
+type SCC struct {
+	// Default contains the default SCC that will be attached to the service
+	// account used for workloads (`pipeline` SA by default) and defined in
+	// PipelineProperties.OptionalPipelineProperties.DefaultServiceAccount
+	// +optional
+	Default string `json:"default,omitempty"`
+	// MaxRequest specifies the highest SCC that can be requested for in a
+	// namespace or in the Default field.
+	// +optional
+	MaxRequest string `json:"maxRequest,omitempty"`
```
Comment (piyush-garg) [inline pkg/apis/operator/v1alpha1/openshift_platform.go:46]: we discussed to change this field, can we please do that change

File: pkg/reconciler/openshift/tektonconfig/rbac.go [hunk 3]
```diff
@@ -19,9 +19,9 @@ package tektonconfig
 import (
 	"context"
 	"fmt"
-	"math"
-	"regexp"
-
+	securityv1 "github.com/openshift/api/security/v1"
```
Comment (piyush-garg) [inline pkg/reconciler/openshift/tektonconfig/rbac.go:22]: can you please use `goimports`

this is validating that and same the case with many other files

File: pkg/reconciler/openshift/tektonconfig/rbac_test.go [hunk 4]
```diff
@@ -0,0 +1,157 @@
+package tektonconfig
```
Comment (piyush-garg) [inline pkg/reconciler/openshift/tektonconfig/rbac_test.go:1]: add license

File: pkg/apis/operator/v1alpha1/openshift_platform.go (hunk 2 above)
Comment (concaf) [reply pkg/apis/operator/v1alpha1/openshift_platform.go]: changed to `maxAllowed`

File: pkg/reconciler/openshift/tektonconfig/rbac.go (hunk 3 above)
Comment (concaf) [reply pkg/reconciler/openshift/tektonconfig/rbac.go:27]: done

File: pkg/reconciler/openshift/tektonconfig/rbac_test.go (hunk 4 above)
Comment (concaf) [reply pkg/reconciler/openshift/tektonconfig/rbac_test.go:17]: done

**Pull request merged**